service/src/main/java/uk/gov/hmcts/reform/fpl/service/MigrateCaseService.java

```diff
@@ -208,4 +209,19 @@ public Map<String, Object> removeCaseNote(CaseData caseData, String migrationId,
         return Map.of("caseNotes", caseNoteService.removeCaseNote(caseNoteIdToRemove, caseData.getCaseNotes()));
     }
 
+    public void verifyGatekeepingOrderUrgentHearingOrder(CaseData caseData, String migrationId, String fileName) {
+        if (caseData.getUrgentHearingOrder() == null) {
+            throw new AssertionError(format(
+                "Migration {id = %s, case reference = %s}, GateKeeping order - Urgent hearing order not found",
+                migrationId, caseData.getId()));
+        }
+
+        if (isNotEmpty(fileName)) {
+            if (!fileName.equals(caseData.getUrgentHearingOrder().getOrder().getFilename())) {
```

Comment:
null checking on getOder()?

Reply:
if we want to check the file name, the Order must not be null. Otherwise, why don't we just ignore the fileName field?

Reply:
perhaps throw an exception if unexpected null in `urgentHearingOrder.order`?